### PR TITLE
:recycle: Remove _misource access from MI_Arrow

### PR DESCRIPTION
**Pull request by canassa**

### CODE REVIEW

**Review by AJenbo — APPROVED** [review]

**Comment by AJenbo** [comment]
I would like to put in a request for InitMissiles, AddTown to be done next

**Comment by canassa** [comment]
> I would like to put in a request for InitMissiles, AddTown to be done next

On it 😃

File: Source/missiles.cpp [hunk 1]
```diff
@@ -2699,23 +2699,24 @@ void MI_Arrow(Missile &missile)
...
-			maxd = player._pIMaxDam;
-		} else {
-			auto &monster = Monsters[p];
-			mind = monster.minDamage;
-			maxd = monster.maxDamage;
-		}
-	} else {
+	switch (missile.sourceType()) {
+	case MissileSource::Player: {
+		const Player *player = missile.sourcePlayer();
```
Comment (glebm) [inline Source/missiles.cpp:2707]: We already check the source type, and then `sourcePlayer/Monster` checks the type again!

Generally, for variant types, the method that returns a particular value is not checked, and the caller instead must check the type. Let's do the same?

Comment (AJenbo) [reply Source/missiles.cpp:2707]: I think this is mostly a temporary issue until everything has been moved over.

Comment (canassa) [reply Source/missiles.cpp:2707]: That means we can also change sourceMonster and sourcePlayer to return a reference instead of a pointer, right?

Comment (glebm) [reply Source/missiles.cpp:2707]: Yeah

**Pull request merged**